gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/IOSStringsResource.java

```diff
@@ -16,112 +16,190 @@
-        StringBuilder temp = new StringBuilder(100);
-        for (String key : map.keySet()) {
-            // empties the buffer
-            temp.setLength(0);
+    public void write(OutputStream os, String language, Collection<ResourceString> resStrings) throws IOException {
+        TreeSet<ResourceString> sortedResources = new TreeSet<>(new ResourceStringComparator());
+        sortedResources.addAll(resStrings);
+
+        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(os, CHAR_SET));
 
-            String value = map.get(key);
-            temp.append(key).append("=").append(value).append("\n");
-            os.write(temp.toString().getBytes(Charset.forName(CHARSET_STRING)));
+        for (ResourceString res : sortedResources) {
+            // empties the buffer
+            writer.write(formatEntry(res.getKey(), res.getValue(), language));
         }
+
+        writer.flush();
     }
 
     @Override
-    public void merge(InputStream base, OutputStream os, String language, Map<String, String> map) throws IOException {
-        Scanner in = new Scanner(base, "UTF-8");
-        String line = "";
-        while (in.hasNextLine()) {
-            line = in.nextLine() + '\n';
-            if (line.indexOf("=") == -1) {
-                os.write(line.getBytes());
-            } else {
-                String key = line.substring(0, line.indexOf("=")).trim();
-                if (map.containsKey(key)) {
-                    StringBuilder temp = new StringBuilder(100);
-                    temp.append("\"").append(key).append("=").append("\"");
-
-                    final int character_offset = 80;
-                    BreakIterator b = BreakIterator.getWordInstance();
-                    b.setText(map.get(key));
-
-                    int offset = 80;
-                    int start = 0;
-
-                    boolean first = true;
-
-                    while (start < map.get(key).length()) {
-                        if (map.get(key).length() > character_offset) {
-                            if (first) {
-                                temp.append("\"");
-                            }
-                            if (!first) {
-                                temp.append(" ");
-                            }
-
-                            first = false;
-                            int end = b.following(offset);
-                            String str = map.get(key).substring(start, end);
-                            start = end;
-                            offset += 80;
-                            temp.append(str).append(" %@\n");
-                        } else {
-                            temp.append("\"").append(map.get(key));
-                            start = map.get(key).length();
-                        }
+    public void merge(InputStream base, OutputStream os, String language, Collection<ResourceString> data)
```

Comment:
is there a reason `merge()` isn't implemented as `parse(A); parse(B); write();`?